### PR TITLE
Make daml start integration tests use canton.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
- Switch to using --auto-connect-local instead of a bootstrap script in daml start and daml canton-sandbox.
- Fixed the ledger id used in the daml start json api header (else it hangs).
- Changed the daml start integration tests to use --sandbox-canton
  - Haven't fixed the hot reload test yet.
- Separated out a daml start test that doesn't use the shared instance.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs [hunk 1]
```diff
@@ -168,12 +168,13 @@ withJsonApi (SandboxPort sandboxPort) (JsonApiPort jsonApiPort) extraArgs a = do
             , "--http-port", show jsonApiPort
             , "--allow-insecure-tokens"
             ] ++ extraArgs
+    putStrLn ("Starting JSON API: " <> unwords args)
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs:171]: did you intentionally leave this in? Seems a bit inconsistent to have this for the JSON API but not for other stuff, e.g., Navigator.

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs [hunk 2]
```diff
@@ -168,12 +168,13 @@ withJsonApi (SandboxPort sandboxPort) (JsonApiPort jsonApiPort) extraArgs a = do
...
     withPlatformJar args "json-api-logback.xml" $ \ph -> do
         putStrLn "Waiting for JSON API to start: "
         -- The secret doesn’t matter here
         let token = JWT.encodeSigned (JWT.HMACSecret "secret") mempty mempty
                 { JWT.unregisteredClaims = JWT.ClaimsMap $
-                      Map.fromList [("https://daml.com/ledger-api", Object $ HashMap.fromList [("actAs", toJSON ["Alice" :: T.Text]), ("ledgerId", "MyLedger"), ("applicationId", "foobar")])]
+                      Map.fromList [("https://daml.com/ledger-api", Object $ HashMap.fromList [("actAs", toJSON ["Alice" :: T.Text]), ("ledgerId", "sandbox"), ("applicationId", "foobar")])]
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs:177]: This is a bit fishy. We really don’t know the ledger id. However in SDK 2.0 the ledger id will become optional. I suggest adding a TODO here to update this and link it to some issue.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 3]
```diff
@@ -8,11 +8,13 @@ import Control.Concurrent.STM
 import Control.Exception.Extra
 import Control.Lens
 import Control.Monad
-import Control.Monad.Loops (untilM_)
+-- import Control.Monad.Loops (untilM_)
```
Comment (cocreature) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs:11]: leftover?

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 4]
```diff
@@ -151,14 +153,18 @@ damlStart tmpDir = do
     waitForHttpServer
         (threadDelay 100000)
         ("http://localhost:" <> show jsonApiPort <> "/v1/query")
-        authorizationHeaders
+        (authorizationHeaders "Alice") -- dummy party here, not important
```
Comment (cocreature) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs:156]: we could instead check `/v1/parties` with an admin token to not have to make up a non-existent party.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 5]
```diff
@@ -481,8 +488,10 @@ damlStartTests getDamlStart =
                             (threadDelay 100000)
                             ("http://localhost:" <> show navigatorPort)
                             []
+
+{-
```
Comment (cocreature) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs:492]: Can we keep testing this against the old sandbox until it works with Canton?

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs (hunk 1 above)
Comment (sofiafaro-da) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs:171]: Yep, oops!

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs (hunk 2 above)
Comment (sofiafaro-da) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs:177]: I was thinking of adding a --ledgerid option in daml start, but if it becomes optional we can just drop it.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs (hunk 3 above)
Comment (sofiafaro-da) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs:11]: no, it's from commenting out the hot reload test, until I can reenable it

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs (hunk 2 above)
Comment (sofiafaro-da) [reply daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs]: Added a TODO and linked to https://github.com/digital-asset/daml/issues/12145

**Review by robin-da — APPROVED** [review]
Lgtm. If possible I'd also like the hot-reload tests running against the old sadbox.

**Comment by sofiafaro-da** [comment]
I re-enabled the hot reload tests against sandbox-kv :-)

**Pull request merged**